enhancements/cluster-logging/forward_to_cloudwatch.md

```diff
@@ -0,0 +1,226 @@
+---
+title: forward_to_cloudwatch
+authors:
+  - "@alanconway"
+reviewers:
+  - "@jcantrill"
+  - "@jeremyeder"
+approvers:
+creation-date: 2020-12-17
+last-updated: 2020-12-17
+status: implementable
+see-also:
+superseded-by:
+---
+
+# Forward to Cloudwatch
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+[Amazon Cloudwatch][aws-cw] is a hosted monitoring and log storage service.
+This proposal extends the `ClusterLogForwarder` API with an output type for Cloudwatch.
+
+## Motivation
+
+Amazon Cloudwatch is a popular log store.
+We have requests from external and  Red Hat-internal customers to support it.
+
+### Goals
+
+Enable log forwarding to Cloudwatch.
+
+### Non-Goals
+
+Enable Cloudwatch metric collection.
+
+## Proposal
+
+### Cloudwatch streams and groups
+
+[Cloudwatch][concepts] defines *log groups* and *log streams*. To paraphrase the documentation:
+
+>  A log stream is a sequence of log events that share the same source ... For example, an Apache access log on a specific host.
+
+>  Log groups define groups of log streams that share the same retention, monitoring, and access control settings ... For example, if you have a separate log stream for the Apache access logs from each host, you could group those log streams into a single log group called MyWebsite.com/Apache/access_log.
+
+In other words a *log stream* corresponds to the smallest log-producing units.
+A *log group* is a collection of related *log streams*.
+
+We must consider *container logs*, and *node logs*.
+- "application" logs are always container logs
+- "infrastructure" logs are a mix of container and node logs
+- "audit" logs are all node logs.
+
+For *container-scoped* logs we auto-create a log stream for *each container*, using the container UUID as the log stream name.
+- The log stream name is just a unique identifier for the source container.
+- Log *entries* include meta-data (container name, namespace etc.) for the user to search/index.
+  The log stream name is *not* needed for indexing or searching logs.
+- Log streams are for a *single source*, we cannot send logs from multiple nodes on the same log stream.
```

Comment:
Not sure this is a "can" or "can not" thing, but more of a "do" or "do not" thing.  We do not *want* to send logs from multiple nodes to the same log stream to avoid hitting hard rate-limits applied to individual log streams.